### PR TITLE
[2.19.x] G-9453 Added CQL Sanitizing method for Search Save and Query Titles

### PR DESCRIPTION
**Pull request by jMoneee**
#### What does this PR do?

Adds a method to sanitize CQL queries for use in downstream projects as well as ported changes from https://github.com/codice/ddf/pull/6433 which solves a related issue.

#### Who is reviewing it? 

@leo-sakh
@jMoneee
@frnkshin
@cassandrabailey293
#### How should this be tested?

build / run, verify no regression when building searches without titles and default titles are getting set /saved correctly, verify when editing a query by removing it's title that saving / running the query produces the correct default title.

#### What are the relevant tickets?
Fixes: #____

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/js/model/Query.js [hunk 1]
```diff
@@ -316,6 +316,9 @@ Query.Model = PartialAssociatedModel.extend({
       })
     })
   },
+  async cqlSanitizePlugin(data) {
```
Comment (cassandrabailey293) [inline ui/packages/catalog-ui-search/src/main/webapp/js/model/Query.js:319]: remove `async`

File: ui/packages/catalog-ui-search/src/main/webapp/component/query-editor/query-editor.view.js [hunk 2]
```diff
@@ -165,9 +170,13 @@ module.exports = Marionette.LayoutView.extend({
     }
     queryContentView.save()
     this.queryTitle.currentView.save()
+    if (this.model.get('title') === '') {
+      this.setDefaultTitle()
+    }
     if (store.getCurrentQueries().get(this.model) === undefined) {
       store.getCurrentQueries().add(this.model)
     }
+    this.model.cqlSanitizePlugin(store.getCurrentQueries())
```
Comment (cassandrabailey293) [inline ui/packages/catalog-ui-search/src/main/webapp/component/query-editor/query-editor.view.js:179]: remove `store.getCurrentQueries()` from this call

File: ui/packages/catalog-ui-search/src/main/webapp/js/model/Query.js (hunk 1 above)
Comment (cassandrabailey293) [reply ui/packages/catalog-ui-search/src/main/webapp/js/model/Query.js:319]: also remove `data` from the method signature, and just `return` (do nothing) in this method. don't need to return `data`

**Review by cassandrabailey293 — APPROVED** [review]

**Review by andrewzimmer — APPROVED** [review]

**Review by lavoywj — APPROVED** [review]

**Comment by lavoywj** [comment]
Hero was successful
- verified no regression when building searches without titles and default titles are getting set /saved correctly
- verified editing a query by removing it's title that saving / running the query produces the correct default title

**Comment by cassandrabailey293** [comment]
build now

**Comment by shaundmorris** [comment]
Merging. Build failed due to known docker issue.

**Pull request merged**